pkg/api/grpc_handlers.go

```diff
@@ -66,6 +66,163 @@ type handlers struct {
 	appInfo         *dom.AppInfo
 }
 
+func (h *handlers) StartConsistencyCheck(ctx context.Context, req *pb.StartConsistencyCheckRequest) (*pb.StartConsistencyCheckResponse, error) {
+	consistencyCheckID := MakeConsistencyCheckID(req.Locations)
+
+	locations := make([]tasks.MigrateLocation, 0, len(req.Locations))
+	for _, location := range req.Locations {
+		locations = append(locations, tasks.MigrateLocation{
+			Storage: location.Storage,
+			Bucket:  location.Bucket,
+			User:    location.User,
+		})
+	}
+
+	consistencyCheckTask := tasks.ConsistencyCheckPayload{
+		ID:        consistencyCheckID,
+		Locations: locations,
+	}
+
+	task, err := tasks.NewTask(ctx, consistencyCheckTask)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create consistency check task: %w", err)
+	}
+	if _, err := h.taskClient.EnqueueContext(ctx, task); err != nil && !errors.Is(err, asynq.ErrDuplicateTask) && !errors.Is(err, asynq.ErrTaskIDConflict) {
+		return nil, fmt.Errorf("unable to enqueue consistency check task: %w", err)
+	}
+
+	return &pb.StartConsistencyCheckResponse{
+		Id: consistencyCheckID,
+	}, nil
+}
+
+func (h *handlers) ListConsistencyChecks(ctx context.Context, _ *emptypb.Empty) (*pb.ListConsistencyChecksResponse, error) {
+	ids, err := h.storageSvc.ListConsistencyCheckIDs(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("unable to list consistency check ids: %w", err)
+	}
+
+	consistencyChecks := make([]*pb.ConsistencyCheck, 0, len(ids))
+	for _, id := range ids {
+		consistencyCheck, err := h.constructConsistencyCheck(ctx, id)
+		if err != nil {
+			return nil, fmt.Errorf("unable to construct consistency check: %w", err)
+		}
+
+		consistencyChecks = append(consistencyChecks, consistencyCheck)
+	}
+
+	return &pb.ListConsistencyChecksResponse{
+		Checks: consistencyChecks,
+	}, nil
+}
+
+func MakeConsistencyCheckID(locations []*pb.MigrateLocation) string {
+	sort.Slice(locations, func(i, j int) bool {
```

Comment:
is it makese sense to add validation?
- len(locations)!=0
- user/storage/bucket !=""
- user/storage presented in config.

here is [example from replication api method](https://github.com/clyso/chorus/blob/66addbd468a9c822efa15bf07094b37be1d47cfb/pkg/api/grpc_handlers.go#L143-L151).

Reply:
i thought that this method is good place for validation because it is used in all  consistency check api methods. 

It is better to fail fast and do validation before task enqueue.

Reply:
Added a method to validate incoming locations. Please verify.